tests/tasks/assert_kernel_settings_values.yml

```diff
@@ -0,0 +1,61 @@
+---
+- name: check that settings are applied correctly
+  command: tuned-adm verify -i
+  ignore_errors: true
+  register: __kernel_settings_register_verify_values
+  changed_when: false
+
+# have to verify bootloader cmdline settings separately
+# "Sometimes (if some plugins like bootloader are used) a
+# reboot may be required."
+# warn - do not warn about the use of sed
+- name: get last verify results from log
+  shell: |
+    set -euo pipefail
+    sed '/INFO     tuned.daemon.daemon: verifying profile/h;//!H;$!d;x' \
+      /var/log/tuned/tuned.log | \
+      grep -v 'ERROR    tuned.plugins.plugin_bootloader: verify: failed' | \
+      grep 'ERROR    tuned.plugins' || :
+  args:
+    warn: false
+  register: __kernel_settings_register_verify_log
+  when: __kernel_settings_register_verify_values.rc != 0
+
+- name: report errors that are not bootloader errors and set error flag
+  set_fact:
+    __kernel_settings_success: false
+  when:
+    - __kernel_settings_register_verify_values.rc != 0
+    - __kernel_settings_register_verify_log is defined
+    - __kernel_settings_register_verify_log.stdout_lines is defined
+    - __kernel_settings_register_verify_log.stdout_lines != []
+
+# warn - do not warn about the use of sed
+- name: get the bootloader specific config file
+  command: >-
+    sed -n '/^TUNED_BOOT_CMDLINE=/{s/^TUNED_BOOT_CMDLINE="\(.*\)"$/\1/;p}'
+    /etc/tuned/bootcmdline
+  args:
+    warn: false
+  register: __kernel_settings_register_verify_bl_cmdline
+  when:
+    - __kernel_settings_register_parameters is defined
+    - __kernel_settings_register_parameters.bootloader is defined
+    - __kernel_settings_register_parameters.bootloader.cmdline is defined
+
+- name: verify bootloader settings command
+  set_fact:
+    __kernel_settings_success: false
+  when:
+    - __kernel_settings_register_verify_bl_cmdline is defined
+    - __kernel_settings_register_verify_bl_cmdline.rc is defined
+    - __kernel_settings_register_verify_bl_cmdline.rc != 0
```

Comment:
there is a test for this, https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html#task-results

Reply:
but will it ever succeed? With a nonzero exist status, how will we reach this task?

Reply:
> but will it ever succeed? With a nonzero exist status, how will we reach this task?

Not sure what you mean?

Reply:
`__kernel_settings_register_verify_bl_cmdline` comes from a previous `command` task 
The previous task does not have any `ignore_errors` or `failed_when`.
Therefore, if `rc` is nonzero, it will fail and we will not reach this task. The test `__kernel_settings_register_verify_bl_cmdline.rc != 0` will never succeed. Am I missing something?

Reply:
Got it - fixed it - also converted the test to work with the bootloader value being a `list` instead of a `dict` (that is, there is no more `parameters.bootloader.cmdline`) - see https://github.com/linux-system-roles/kernel_settings/pull/1/commits/0ac194a33e6976d11a7baae252e848797f875a21